core/jobs/batch_jobs/opportunity_management_jobs.py

```diff
@@ -337,14 +337,15 @@ def _generate_opportunities_related_to_topic(
             stories = [
                 stories_dict[story_id] for story_id in existing_story_ids
             ]
-            for story in stories:
-                for exp_id in story.story_contents.get_all_linked_exp_ids():
-                    exploration_opportunity_summary_list.append(
-                        opportunity_services.create_exp_opportunity_summary(
-                            topic, story, exps_dict[exp_id]))
 
             exploration_opportunity_summary_model_list = []
             with datastore_services.get_ndb_context():
+                for story in stories:
+                    for exp_id in story.story_contents.get_all_linked_exp_ids():
+                        exploration_opportunity_summary_list.append(
+                            opportunity_services.create_exp_opportunity_summary(
+                                topic, story, exps_dict[exp_id]))
+
```

Comment:
Why is this moved?

Reply:
As `create_exp_opportunity_summary` will require fetching translation models to count the number of translations available.